pkg/webhook/server/server.go

```diff
@@ -139,6 +173,12 @@ func (s *Server) Run(ctx context.Context) error {
 		return err
 	}
 
+	if s.MetricsCertificateSource != nil {
+		if err := mgr.Add(s.MetricsCertificateSource); err != nil {
+			return err
+		}
+	}
```

Comment:
This starts another TLS certificate loop for the metrics server, watches the CA Secret if dynamic certificate source has been configured.